src/test/java/persistence/sql/ddl/DDLTest.java

```diff
@@ -0,0 +1,88 @@
+package persistence.sql.ddl;
+
+import database.DatabaseServer;
+import database.H2;
+import domain.PersonV1;
+import domain.PersonV2;
+import domain.PersonV3;
+import jdbc.JdbcTemplate;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.*;
+import persistence.dialect.Dialect;
+import persistence.dialect.H2Dialect;
+import sources.AnnotationBinder;
+import sources.MetaData;
+import sources.MetadataGenerator;
+import sources.MetadataGeneratorImpl;
+
+import java.sql.SQLException;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class DDLTest {
+    DatabaseServer server;
+    final Dialect dialect = new H2Dialect();
+    final QueryBuilder queryBuilder = new QueryBuilder(dialect);
+    final AnnotationBinder annotationBinder = new AnnotationBinder();
+    final MetadataGenerator metadataGenerator = new MetadataGeneratorImpl(annotationBinder);
+    JdbcTemplate jdbcTemplate;
+
+    @BeforeAll
+    void setUp() throws SQLException {
+        this.server = new H2();
+        server.start();
+        this.jdbcTemplate = new JdbcTemplate(server.getConnection());
+    }
+
+    @Test
+    @DisplayName("요구사항 1 - 아래 정보를 바탕으로 create 쿼리 만들어보기")
+    void createTest1() {
+        MetaData personv1 = metadataGenerator.generator(PersonV1.class);
+        StringBuilder sb = new StringBuilder();
+        StringBuilder createQuery = queryBuilder.create(personv1, sb);
+        jdbcTemplate.execute(String.valueOf(createQuery));
+        Assertions.assertThat("create table PersonV1 (id INT AUTO_INCREMENT PRIMARY KEY , name varchar , age int  )")
+                .isEqualTo(String.valueOf(createQuery));
+    }
+
+    @Test
+    @DisplayName("요구사항 2 - 추가된 정보를 통해 create 쿼리 만들어보기")
+    void createTest2() {
+        MetaData personv2 = metadataGenerator.generator(PersonV2.class);
+        StringBuilder sb = new StringBuilder();
+        StringBuilder createQuery = queryBuilder.create(personv2, sb);
+        jdbcTemplate.execute(String.valueOf(createQuery));
+        System.out.println(String.valueOf(createQuery));
+        Assertions.assertThat("create table PersonV2 (id INT AUTO_INCREMENT PRIMARY KEY , nick_name varchar , old int , email varchar not null )")
+                .isEqualTo(String.valueOf(createQuery));
+    }
+
+    @Test
+    @DisplayName("요구사항 3 - 추가된 정보를 통해 create 쿼리 만들어보기")
+    void createTest3() {
+        MetaData personv3 = metadataGenerator.generator(PersonV3.class);
+        StringBuilder sb = new StringBuilder();
+        StringBuilder createQuery = queryBuilder.create(personv3, sb);
+        jdbcTemplate.execute(String.valueOf(createQuery));
+        System.out.println(String.valueOf(createQuery));
+        Assertions.assertThat("create table users (id INT AUTO_INCREMENT PRIMARY KEY , nick_name varchar , old int , email varchar not null )")
+                .isEqualTo(String.valueOf(createQuery));
```

Comment:
```suggestion
        Query createQuery = queryBuilder.create(person);
        ...
        Assertions.assertThat("....")
                .isEqualTo(createQuery.toQuery());
```

저는 검증하고자하는 메소드의 결과를 추가 가공하는것은 지양하는 편입니다. 
별도의 쿼리를 자료로 가지는 객체가 있다면 SQL 사용에 대한 오용을 방지 할 수 있지 않을까요?

참고 : [원시값 포장](https://00h0.tistory.com/79)